db/dcrpg/upgrades.go

```diff
@@ -401,8 +389,8 @@ func (pgb *ChainDB) CheckForAuxDBUpgrade(dcrdClient *rpcclient.Client) (bool, er
 	}
 
 	// Ensure the required version was reached.
-	upgradeInfo = TableUpgradesRequired(TableVersions(pgb.db))
-	if len(upgradeInfo) > 0 && upgradeInfo[0].UpgradeType != "ok" {
+	upgradeInfo := TableUpgradesRequired(TableVersions(pgb.db))
+	if len(upgradeInfo) > 0 && upgradeInfo[0].UpgradeType != OK {
```

Comment:
This is still randomly ordered because `TableUpgradesRequired` loops over the map returned by `TableVersions`. If the first table in the loop is OK, then there will still be no upgrade. This needs to check if any of the tables require an upgrade.